### PR TITLE
[tvla] Enable TVLA general-data analysis

### PR DESCRIPTION
**Pull request by vrozic**
This commit enables TVLA fixed-vs-random data analysis. In addition, this commit re-names related functions and variables to more accurate and intuitive names.

The main change is in `tvla.py`
The remaining changes are mostly in configuration files due to changing parameter from boolean `general_test` to string `test_type` in `"GENERAL_KEY", "GENERAL_DATA", "SPECIFIC"`.

### CODE REVIEW

File: analysis/tvla.py [hunk 1]
```diff
@@ -488,7 +510,7 @@ def run_tvla(ctx: typer.Context):
                              trace_start=trace_start, trace_end=trace_end)
 
                 if ((save_to_disk_trace is True or save_to_disk_ttest is True) and
-                        general_test is True and i_step == 0):
+                        (general_test) and i_step == 0):
```
Comment (vogelpi) [inline analysis/tvla.py:513]: Do you need the parenthesis around `general_test` here?

File: analysis/tvla.py [hunk 2]
```diff
@@ -585,17 +614,22 @@ def run_tvla(ctx: typer.Context):
...
-            else:
+            elif general_test_key:
                 log.info("Computing Leakage")
                 # We identify the fixed key by looking at the first 20 keys in the project.
-                leakage = compute_leakage_general(keys, find_fixed_key(keys_nparrays[0:20]))
+                leakage = compute_leakage_general(keys, find_fixed_entry(keys_nparrays[0:20]))
+            else:
+                assert general_test_data
+                log.info("Computing Leakage")
+                # We identify the fixed key by looking at the first 20 plaintexts in the project.
```
Comment (vogelpi) [inline analysis/tvla.py:624]: Nit: "We identify the fixed data..."

File: analysis/tvla.py [hunk 3]
```diff
@@ -585,17 +614,22 @@ def run_tvla(ctx: typer.Context):
...
                     leakage = np.load(cfg["leakage_file"])
                     assert num_traces == leakage.shape[2]
-            else:
+            elif general_test_key:
                 log.info("Computing Leakage")
                 # We identify the fixed key by looking at the first 20 keys in the project.
-                leakage = compute_leakage_general(keys, find_fixed_key(keys_nparrays[0:20]))
+                leakage = compute_leakage_general(keys, find_fixed_entry(keys_nparrays[0:20]))
+            else:
+                assert general_test_data
```
Comment (vogelpi) [inline analysis/tvla.py:622]: Good idea to add this assertion here.

**Review by vogelpi — APPROVED** [review]
This looks good, I did a thorough review. Thanks @vrozic !

File: analysis/tvla.py (hunk 1 above)
Comment (vrozic) [reply analysis/tvla.py]: No. I've removed them now

File: analysis/tvla.py (hunk 2 above)
Comment (vrozic) [reply analysis/tvla.py]: Thanks for catching this. It is fixed now.

**Pull request merged**